plugins/bundle/plugin.go

```diff
@@ -258,6 +258,12 @@ func (p *Plugin) oneShot(ctx context.Context, name string, u download.Update) {
 
 func (p *Plugin) process(ctx context.Context, name string, u download.Update) {
 
+	if u.Metrics != nil {
+		p.status[name].Metrics = u.Metrics
+	} else {
+		p.status[name].Metrics = metrics.New()
```

Comment:
Out of curiosity, when does this occur?

Reply:
Mostly in unit tests, but indicated that it was easy enough for anyone consuming the package to provide an update with a nil metric. Instead of being defensive other places we used it while processing the update I followed the same pattern we use in like the `rego` package of inserting one and them assuming it exists elsewhere.